Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -674,7 +675,7 @@ namespace Babylon
 
         vertexBufferData->EnsureFinalized(info.Env(), vertexLayout);
 
-        vertexArray.vertexBuffers.push_back({vertexBufferData, byteOffset / byteStride, bgfx::createVertexLayout(vertexLayout)});
+        vertexArray.vertexBuffers[location] = {vertexBufferData, byteOffset / byteStride, bgfx::createVertexLayout(vertexLayout)};
```

Comment:
the only time I can see this is if we have a shader with duplicate attributes with identical names, which will probably fail compilation, otherwise we seem to ensure that we assign unique locations to each attribute, would it be a good idea to add an assertion to ensure this?

Reply:
use `.insert` instead of `[]` and it will return if the entry already exists.